src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientImpl.java

```diff
@@ -371,4 +382,26 @@ private HttpUriRequest buildRequest(GraphqlRequest request, RequestOptions optio
 
         return rb.build();
     }
+
+    static class ConfigurableConnectionKeepAliveStrategy implements ConnectionKeepAliveStrategy {
+        private int defaultConnectionKeepAlive;
+
+        public ConfigurableConnectionKeepAliveStrategy(int defaultConnectionKeepAlive) {
+            this.defaultConnectionKeepAlive = defaultConnectionKeepAlive;
+        }
+
+        @Override
+        public long getKeepAliveDuration(HttpResponse httpResponse, HttpContext httpContext) {
+            HeaderElementIterator it = new BasicHeaderElementIterator(httpResponse.headerIterator(HTTP.CONN_KEEP_ALIVE));
+            while (it.hasNext()) {
+                HeaderElement he = it.nextElement();
+                String param = he.getName();
+                String value = he.getValue();
+                if (value != null && param.equalsIgnoreCase("timeout")) {
+                    return Long.parseLong(value) * 1000;
+                }
+            }
+            return defaultConnectionKeepAlive * 1000L;
```

Comment:
Shouldn't the semantic rather be a min(configured timeout, returned timeout)? What if the server returns 10m keep alive timeout header parameter but the infrastructure already drops the connection after 5m? 

If so the default should be Integer.MAX_VALUE instead of `-1`

Reply:
Good point. I've implemented it.